backend/models/rom.py

```diff
@@ -421,6 +423,48 @@ class RomUserStatus(enum.StrEnum):
     NEVER_PLAYING = "never_playing"  # Will never play
 
 
+class RomNote(BaseModel):
+    __tablename__ = "rom_notes"
+    __table_args__ = (
+        UniqueConstraint(
+            "rom_id", "user_id", "title", name="unique_rom_user_note_title"
+        ),
+        Index("idx_rom_notes_public", "is_public"),
+        Index("idx_rom_notes_rom_user", "rom_id", "user_id"),
+        Index("idx_rom_notes_title", "title"),
+    )
+
+    id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
+
+    # Core note fields
+    title: Mapped[str] = mapped_column(String(400))
+    content: Mapped[str] = mapped_column(Text)
```

Comment:
this is good, and we can move this content to a physical file at some point via a migration, if we want to write to disk each time